File: assign1/p2.py
def f(x):

    return x ** 4 - 13


def bisection_method(a, b, tolerance=1e-6):    

    iteration = 0
    prev_c = None

    while True:
        iteration += 1
        c = (a + b) / 2
        fc = f(c)

        # คำนวณ error
        if prev_c is not None:
            error = abs(c - prev_c)
        else:
            error = float('inf')

        # ตรวจสอบเงื่อนไขการหยุด (ไม่มีการเปลี่ยนแปลงจุดทศนิยม 6 ตำแหน่ง)
        if prev_c is not None and abs(c - prev_c) < tolerance:
            break

        # ตรวจสอบว่าได้คำตอบแล้วหรือไม่
        if abs(fc) < tolerance:
            print(f"Found root: c = {c:.6f}, f(c) = {fc:.10f}")
            break

        # อัปเดตขอบเขต
        if f(a) * fc < 0:
            b = c
        else:
            a = c

        prev_c = c

        # ป้องกันลูปไม่สิ้นสุด
        if iteration > 100:
            break

    print(f"f({c:.6f}) = {f(c):.10f}")
    print(f"Iterations: {iteration}")

    return c


result = bisection_method(1.5, 2.0)

File: assign1/p_2.cpp
#include <iostream>
#include <math.h>

using namespace std;

double f(double x) {
    return pow(x,4) - 13;
}

double bisection_method(double a, double b, double tolerance) {
    double fa = f(a);
    double fb = f(b);
    if ((fa * fb) > 0) {
        return 0;
    }
    double c;
    do {
        c = (a + b) / 2;
        double fc = f(c);

        if (f(a) * fc < 0) {
            b = c;
        } else {
            a = c;
        }
    } while (abs(a - b) > tolerance);

    return  c;
}

int main(int argc, char *argv[]) {
    double root = bisection_method(1.5, 2, 1e-6);
    cout << root;
}
